hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -242,7 +291,15 @@ public OffsetRange[] getNextOffsetRanges(Option<String> lastCheckpointStr, long
       throw new HoodieException("sourceLimit should not be less than the number of kafka partitions");
     }
 
-    return CheckpointUtils.computeOffsetRanges(fromOffsets, toOffsets, numEvents);
+    OffsetRange[] ranges = CheckpointUtils.computeOffsetRanges(fromOffsets, toOffsets, numEvents);
+    LOG.info("before split by count: " + CheckpointUtils.offsetsStringfy(ranges));
+    long maxEventsPerPartition = props.getLong(KafkaSourceConfig.MAX_EVENTS_PER_KAFKA_PARTITION.key(),
+            KafkaSourceConfig.MAX_EVENTS_PER_KAFKA_PARTITION.defaultValue());
+    LOG.info("getNextOffsetRanges set config " + KafkaSourceConfig.MAX_EVENTS_PER_KAFKA_PARTITION.key() + " to " + maxEventsPerPartition);
+
+    ranges = CheckpointUtils.splitRangesByCount(ranges, maxEventsPerPartition);
```

Comment:
Can we move the split ranges logic to be under CheckpointUtils.computeOffsetRanges as we do a merge there as well in the allocation logic. 

 See the section below the comment in CheckpointUtils
```
// Allocate the remaining events to non-exhausted partitions, in round robin fashion
```

Reply:
sure, It's better to keep such merging and splitting logic together